src/App.tsx

```diff
@@ -1,14 +1,45 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+const initialTodo: Todo = {
+  userId: 0,
+  id: 0,
+  title: '0',
+  completed: false,
+};
+
+let allTodos: Todo[] = [];
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [todo, setTodo] = useState<Todo>(initialTodo);
+  const [loading, setLoading] = useState(true);
+  const [iconEyeId, setIconEyeId] = useState(0);
+
+  function getAllTodos() {
+    getTodos()
+      .then(resultTodos => {
+        allTodos = resultTodos;
+        setTodos(resultTodos);
+      })
+      .finally(() => setLoading(false));
+  }
```

Comment:
The `getAllTodos` function directly modifies the global `allTodos` variable. It's better to manage this state within the component to avoid unintended side effects and improve maintainability.